concepts/modularization/modularization_overview.md

```diff
@@ -0,0 +1,42 @@
+---
+title: Modularization in Kyma
+---
+
+With Kyma’s modular approach, you can install exactly the Kyma components you want, making the installation very lightweight and adjusted to your business needs. Kyma provides three components to support the modular approach: The Module Manager, the Lifecycle Manager, and the Runtime Watcher.
+
+## Kyma’s Modular Approach
+
+![](assets/modular_approach.svg)
+
+## 1. Kyma Module
+
+A Kyma module contains all information required to install and run the associated components in a Kyma runtime:
+
+* A renderable Kubernetes manifest that includes the Kubernetes resources of the module
+* A default configuration for module chart resource installation
+* Optionally, further information
+
+All these assets are bundled into a single container image using the OCI image specification.
+
+> **TIP:** Learn how to build a module(ADD_LINK).
+
+## 2. Module Manager
+
+The Module Manager installs, uninstalls, and manages the Kyma modules in a local (single cluster mode) or remote setup. For example, it remotely manages and synchronizes Kyma module manifest resources in the SAP Kyma Runtime.
+
+The Module Manager retrieves the image layers from the specified image registry and processes the manifest resources by calling a rendering framework like Helm or Kustomize.
+
+Finally, it deploys this resource along with a custom resource to track state changes. It ensures consistency both with a time-based reconciliation and by watching state changes of the deployed custom resource in Kyma runtime.
+
+## 3. Runtime Watcher
+
+The Runtime Watcher monitors the relevant Kyma resources of the user’s Kyma runtime for configured changes, specified by the operator in the control plane, such as:
+
+* Expected changes, like a Kyma user providing a configuration update, which must lead to an update of the particular Kyma runtime.
+* Unexpected changes, like an accidental creation, deletion, or modification, which must be reverted to recover the Kyma runtime to a healthy state.
+
+If needed, the Runtime Watcher triggers a reconciliation of the user’s Kyma runtime by propagating the desired changes to the Module Manager and the Lifecycle Manager.
+
+## 4. Lifecycle Manager
+
+The Lifecycle Manager is responsible for orchestrating Kyma module operators to process their respective resources on the Kyma runtime. This is done by generating the required Kyma module custom resources, which bootstrap Kyma module processing. Furthermore, the Lifecycle Manager reconciles a Kyma custom resource for each Kyma runtime, indicating the consolidated status of all modules configured for that Kyma cluster.
```

Comment:
```suggestion
The Lifecycle Manager is responsible for orchestrating Kyma module operators to process their respective resources on the Kyma runtime. It also aggregates all module states into one easy-to-monitor resource, the `Kyma CR`.

This is done by 

1. generating the required Kyma module custom resources through a Manifest, which starts the Kyma module processing in Module Manager. 
2. reconciling a Kyma custom resource for each Kyma runtime, indicating the consolidated status of all modules configured for that Kyma cluster.

It should be noted that Lifecycle Manager is also responsible to propagate updates from a Module into the respective Manifest through the `ModuleTemplate`. This is a Custom Resource that is used by Lifecycle Manager as a scaffolding when initializing new Modules. It is also used as a base Output of the Module Bundling Process inside the CLI.

Whenever a new Module needs to be installable / registered in the Ecosystem, one only has to apply a new ModuleTemplate with the necessary information for generating the Module. Then it will be available for consumption in every Runtime Instance registered in the Control-Plane.

For every new module, Lifecycle Manager tracks the available ModuleTemplates in different release configurations and makes them available in a Catalog, which allows a browsable interface for all available modules. This Catalog is also synchronized into all Runtime Clusters so users can enable Modules as Runtime Administrators without needing Access to the Control Plane.

Whenever Modules are Updated and a new Release is triggered, Lifecycle Manager will make sure to propagate the correct upgrade information to the Manifest, which in turn leads to updates in the Runtime Cluster.
```

Reply:
Are the lines numbered with `1.` and `2.` subsequent steps, alternatives to each other (unlikely), or concurring processes?

```suggestion
Lifecycle Manager is responsible for orchestrating Kyma module operators to process their respective resources on the Kyma runtime. It also aggregates all module states into one easy-to-monitor resource, the `Kyma CR`, with the following strategies:

1. Generating the required Kyma module custom resources through a manifest, which starts the Kyma module processing in Module Manager. 

2. Reconciling a Kyma custom resource for each Kyma runtime, indicating the consolidated status of all modules configured for that Kyma cluster.

Lifecycle Manager is also responsible for propagating updates from a module into the respective manifest through the ModuleTemplate. Whenever modules are updated and a new release is triggered, Lifecycle Manager propagates the correct upgrade information to the manifest, which in turn leads to updates in the runtime cluster.

## 5. Module Template

The ModuleTemplate is a custom resource that is used by Lifecycle Manager as a scaffolding when initializing new Modules. It is also used as a base output of the module bundling process inside the CLI.

Whenever a new module must be installable or registered in the ecosystem, users only have to apply a new ModuleTemplate with the necessary information for generating the module. Then, it is available for consumption in every runtime instance registered in the control plane.

For every new module, Lifecycle Manager tracks the available ModuleTemplates in different release configurations and makes them available in a catalog, so users have a browsable interface for all available modules. This catalog is also synchronized into all runtime clusters so users can enable modules as runtime administrators without needing access to the control plane.
```